### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Sibusiso-Retro**
Updating Working version  Dev with master

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 1]
```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
...
 
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
         [HttpPost]
-        public virtual void Post([FromBody] User user) {
+        public async Task<ActionResult> Post([FromBody] User user)
```
Comment (guandutoit) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: there are no asynchronous calls being made here please fix.

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 2]
```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
...
+            return new OkObjectResult("Registration successfull, Membership number will be assigned by Admin");
         }
 
         // PUT: api/User/5
         [HttpPut]
-        public void Put([FromBody] User user) {
-            _context.Users.Update(user);
-            _context.SaveChanges();
-        }
+        public async Task<ActionResult> Put([FromBody] User user)
```
Comment (guandutoit) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: no asynchronous calls are being made here

File: SkietbaanBE/SkietbaanBE/Models/ModelsContext.cs [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-using System.Linq;
-using System.Threading.Tasks;
 
 namespace SkietbaanBE.Models {
     public class ModelsContext : DbContext {
         public ModelsContext(DbContextOptions<ModelsContext> options)
            : base(options) { }
 
-        public ModelsContext() : base() { }
+        //public ModelsContext() : base() { }
```
Comment (guandutoit) [inline SkietbaanBE/SkietbaanBE/Models/ModelsContext.cs]: ghost code

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs (hunk 1 above)
Comment (francoisscheffer) [reply SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: Hi, I would also like to see something in this line on your post
if (ModelState.IsValid)
            {
                // Do something with the product (not shown).

                return new HttpResponseMessage(HttpStatusCode.OK);
            }
            else
            {
                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, ModelState);
            }
This is boiler plate so need love but you should get the idea

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 4]
```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
...
+        public async Task<IEnumerable<User>> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
 
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
+        public async Task<User> Get(int id)
```
Comment (guandutoit) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: there are no asynchronous calls being made here please fix.

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 5]
```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
...
+            {
+                return new BadRequestObjectResult("user cannot be null");
+            }
+            else if (Get(user.Id) != null)
+            {
+                return new OkObjectResult("user already exists");
+            }
+
+            //get today's date and save it under user entry date
+            DateTime currentDate = new DateTime();
```
Comment (francoisscheffer) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: Why declare a variable that is never realy used , simple asign to the user object in second line

File: SkietbaanBE/SkietbaanBE/Models/Competition.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SkietbaanBE.Models {
+    public class Competition {
+        public int Id { get; set; }
```
Comment (guandutoit) [inline SkietbaanBE/SkietbaanBE/Models/Competition.cs]: where are your constraints? Is this supposed to be non relational?

File: SkietbaanBE/SkietbaanBE/Models/ModelsContext.cs (hunk 3 above)
Comment (francoisscheffer) [inline SkietbaanBE/SkietbaanBE/Models/ModelsContext.cs]: why the ghost code?

**Comment by Sibusiso-Retro** [comment]
@francoisscheffer , do i have to return the model state, or the badRequest a

**Pull request closed without merging**